### ISSUE TITLE
Outdated Windows build instructions strawberry-msvc/README.md

### ISSUE THREAD

**Issue by Erriez** [strawberrymusicplayer/strawberry-msvc#4]
**Describe the bug**
The Windows build instruction is outdated and broken:

The [README.md](https://github.com/strawberrymusicplayer/strawberry-msvc/blob/master/README.md) refers to `copy-deps-msvc-x64-debug-qt6.bat` and `copy-deps-msvc-x64-release-qt6.bat` which are removed in this commit [Remove copy deps files](https://github.com/strawberrymusicplayer/strawberry-msvc/commit/47dd966512acacb1e7d10433a7202b8332d12dff).

However, these build are required before starting the NSIS setup build. Additionally, several dependencies are missing.

**To Reproduce**
Follow instructions `README.md`.

**Expected behavior**
Clear build description for Windows, no errors.

**Screenshots:**
N.A.

**System Information:**
 - Operating system: Windows
 - Strawberry Version: github.com/strawberrymusicplayer/strawberry-msvc 367396afc61f9731b34c98a935d7b98be70eaa6f

**Additional context**
N.A.

**Comment by jonaski** [comment]
1. What removed bat files are still referenced?
2. What fails exactly?
3. The bat files I deleted were outdated and not used, they were for copying files before creating the installer, creating a installer is not needed for setting up strawberry with Visual Studio for development which is repository is for.

**Comment by Erriez** [comment]
1. Removed `.bat` files is resolved in updated README.md commit e06d17b28f58740e81ed984d714ec93c5e0043e2. (Thanks!)

Open issues `README.md`:
- [This line](https://github.com/strawberrymusicplayer/strawberry-msvc/blob/ea0c4c786691ffd410cd98e248e84f2ba75e7ac5/README.md?plain=1#L69) should be either `build.bat`  or `build.bat debug` without double quotes to build the debug version.
- [This line](https://github.com/strawberrymusicplayer/strawberry-msvc/blob/ea0c4c786691ffd410cd98e248e84f2ba75e7ac5/README.md?plain=1#L74) should be `build.bat release` without double quotes to build the release version.

---
2. Found one root cause for a hidden reproducible issue after version updates:

Several weeks ago:
- Run download.bat -> Downloads ICU 75 source ZIP
- Run build.bat release
  - This installs ICU 75
- ICU version was updated to 76.1 in `version.bat` 
- Run download.bat
- Run build.bat -> Skips new ICU version as the `icu` directory already exists

```
.\build.bat release
...

[ 63%] Linking C shared library libxml2.dll
LINK: command "C:\PROGRA~1\MICROS~2\2022\COMMUN~1\VC\Tools\MSVC\1441~1.341\bin\Hostx86\x64\link.exe /nologo @CMakeFiles\LibXml2.dir\objects1.rsp /out:libxml2.dll /implib:libxml2.lib /pdb:C:\data\projects\strawberry\msvc_\build_release\libxml2-v2.13.4\build\libxml2.pdb /dll /version:2.13 /machine:x64 /INCREMENTAL:NO ws2_32.lib bcrypt.lib C:\strawberry_msvc_x86_64_release\lib\icudt.lib C:\Program Files (x86)\Windows Kits\10\Lib\10.0.22621.0\um\x64\icuin.Lib C:\Program Files (x86)\Windows Kits\10\Lib\10.0.22621.0\um\x64\icuuc.lib C:\strawberry_msvc_x86_64_release\lib\lzma.lib C:\strawberry_msvc_x86_64_release\lib\z.lib C:\strawberry_msvc_x86_64_release\lib\libiconv.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST:EMBED,ID=2" failed (exit code 1120) with the following output:
   Creating library libxml2.lib and object libxml2.exp
encoding.c.obj : error LNK2019: unresolved external symbol UCNV_FROM_U_CALLBACK_STOP_75 referenced in function openIcuConverter
encoding.c.obj : error LNK2019: unresolved external symbol UCNV_TO_U_CALLBACK_STOP_75 referenced in function openIcuConverter
encoding.c.obj : error LNK2019: unresolved external symbol ucnv_open_75 referenced in function openIcuConverter
encoding.c.obj : error LNK2019: unresolved external symbol ucnv_close_75 referenced in function closeIcuConverter
encoding.c.obj : error LNK2019: unresolved external symbol ucnv_setToUCallBack_75 referenced in function openIcuConverter
encoding.c.obj : error LNK2019: unresolved external symbol ucnv_setFromUCallBack_75 referenced in function openIcuConverter
encoding.c.obj : error LNK2019: unresolved external symbol ucnv_convertEx_75 referenced in function xmlCharEncInput
libxml2.dll : fatal error LNK1120: 7 unresolved externals
NMAKE : fatal error U1077: '"C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe" -E vs_link_dll --intdir=CMakeFiles\LibXml2.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x86\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x86\mt.exe --manifests -- C:\PROGRA~1\MICROS~2\2022\COMMUN~1\VC\Tools\MSVC\1441~1.341\bin\Hostx86\x64\link.exe /nologo @CMakeFiles\LibXml2.dir\objects1.rsp @C:\Users\user\AppData\Local\Temp\nmB60B.tmp' : return code '0xffffffff'
Stop.
NMAKE : fatal error U1077: '"C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\bin\HostX86\x64\nmake.exe" -s -f CMakeFiles\LibXml2.dir\build.make /nologo -SL                 CMakeFiles\LibXml2.dir\build' : return code '0x2'
Stop.
NMAKE : fatal error U1077: '"C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\bin\HostX86\x64\nmake.exe" -s -f CMakeFiles\Makefile2 /nologo -LS                 all' : return code '0x2'
Stop.
```

The `libxml2` link failed as CMake found an old installed version ICU 75.

Reason is that this line in [build.bat](https://github.com/strawberrymusicplayer/strawberry-msvc/blob/ea0c4c786691ffd410cd98e248e84f2ba75e7ac5/build.bat#L547) only extracts a new version `icu4c-%ICU4C_VERSION_UNDERSCORE%-src.zip` when the directory `icu` in the build directory does not exist. It does not check which ICU version is installed.

Quick workaround is to remove all output build directories and run the `download.bat` and `build.bat` from scratch (which takes many hours).

Do you have a suggestion how to use the scripts correctly in `README.md`?

---
3. I assume the new build workflows are:
    1. Run `build.bat` without `C:\strawberry_msvc_x86_64_<debug|release>` to build all [strawberry-msvc-dependencies/releases](https://github.com/strawberrymusicplayer/strawberry-msvc-dependencies/releases) from source (very slow), or:
    2. Download binaries from the releases and then re-run `build.bat` which should not rebuild everything from scratch

Build restarted and still running...

**Comment by Erriez** [comment]
Now testing updated [README.md](https://github.com/Erriez/strawberry-msvc/tree/staging/erriez/update-readme/README.md). Stay tuned.

**Comment by Erriez** [comment]
All review comments PR #7 processed, ready for review.
